core/core.go

```diff
@@ -121,22 +123,30 @@ func (core *Core) Setup(sdkSettings model.SdkSettings, deviceProperties model.De
 	done := make(chan struct{})
 	go func() {
 		defer close(done)
-
-		<-core.Fetch()
+		select {
```

Comment:
in general not sure we need a select here
the setup/shutdown on the rox level is locked
so core should not have shutdown and setup mixed, and core won't be referenced anymore after shutdown

am i missing something?